src/app/shared/components/button/template.html

```diff
@@ -29,6 +30,7 @@
 </button>
 <button *ngIf="iconButton"
         mat-icon-button
+        [id]="this.buttonId"
```

Comment:
Applying the same id to multiple elements is invalid HTML and should be avoided.

Reply:
but the tow element wont show together it's either one of them is shown 
you mean line 33 and line 19 right ?there is *ngIf that prevent them to been shown together

Reply:
i thought maybe at some place we would have the km-button component using multiple time in the same page so they would have the same id if it's constant and in the tests we need to get specific button by id  to click the button inside the component so if they have the same id both will be clicked